src/module.c

```diff
@@ -7532,6 +7533,32 @@ ValkeyModuleString *VM_SaveDataTypeToString(ValkeyModuleCtx *ctx, void *data, co
     }
 }
 
+/**
+ * Serialize the value of a given key from the database of the module context's client.
+ * If the serialization fails for any reason (e.g. key doesn't exist, or rdbSaveObject() fails), return NULL
+ * Otherwise return ValkeyModuleString which contains the serialized value for the key.
+ */
+ValkeyModuleString *VM_SaveObjectToString(ValkeyModuleCtx *ctx, ValkeyModuleString *key) {
+    if (!ctx || !key) {
+        return NULL;
+    }
+
+    int flags = LOOKUP_NOTOUCH | LOOKUP_NONOTIFY | LOOKUP_NOEXPIRE | LOOKUP_NOEFFECTS;
+    robj *value = lookupKeyReadWithFlags(ctx->client->db, key, flags);
+    if (value == NULL) {
+        return NULL;
+    }
+    rio rdb;
+    rioInitWithBuffer(&rdb, sdsempty());
+    createDumpPayload(&rdb, value, key, ctx->client->db->id);
+
+    // Create a ValkeyModuleString (robj*) with the result SDS and return it.
+    sds result = rdb.io.buffer.ptr;
+    ValkeyModuleString *o = createObject(OBJ_STRING, result);
+    if (ctx != NULL) autoMemoryAdd(ctx, VALKEYMODULE_AM_STRING, o);
```

Comment:
on the entry of the method we check if ctx is NULL, we perform an early exit so we could avoid the NULL check here.

```suggestion
    autoMemoryAdd(ctx, VALKEYMODULE_AM_STRING, o);
```